### DISCUSSION TITLE
Where should the Toaster component be placed in NextJS apps

### DISCUSSION THREAD

**Discussion by obrassard** [timolins/react-hot-toast#245] [Q&A] [answered]
Hello, I am trying to use react-hot-toast in a Next JS app, and I wonder if someone knows a way to add the `<Toaster />` in a page template (in `pages/`) or in `_document.jsx`. Indeed, placing it there doesn't seem to work properly, since no toast is displayed when using the `toast()` function, probably because of a SSR issue...

The only way I succeed to display a toast was to place `<Toaster />` in the component where I call `toast()` (a form component) down in the component hierarchy .

**Comment by digvalley** [comment, marked as answer]
You can put it in  _app.jsx
